### ISSUE TITLE
Type-on-hover for type constructors shows linear arrows even if -XLinearTypes is not enabled

### ISSUE THREAD

**Issue by akrmn** [haskell/haskell-language-server#2904]
This is probably confusing for newcomers who don't know about `LinearTypes`, even if technically correct according to the user's guide 

> By default, all fields in algebraic data types are linear 
> -- https://downloads.haskell.org/ghc/9.0.2/docs/html/users_guide/exts/linear_types.html#data-types

GHCi gets around this because, also according to the user's guide,

> [...] Given `data T1 a = MkT1 a` [...] [w]hen used as a value, `MkT1` is given a multiplicity-polymorphic type: `MkT1 :: forall {m} a. a %m -> T1 a`. [...] When displaying types, unless -XLinearTypes is enabled, multiplicity polymorphic functions are printed as regular functions [...]; therefore constructors appear to have regular function types.
> -- https://downloads.haskell.org/ghc/9.0.2/docs/html/users_guide/exts/linear_types.html#data-types

The solution, then, would be to use `MkT1`-the-value for the type-on-hover tooltip, instead of `MkT1`-the-constructor. 

### Your environment

Which OS do you use: macOS
Which LSP client (editor/plugin) do you use: VSCode
Describe your project (alternative: link to the project): plain cabal project, ghc-9.0.2

### Steps to reproduce
Define a newtype or data type and hover on one of its constructors.

```haskell
newtype WrappedInt = MkWrapped Int
```

### Expected behaviour
The tooltip shows the type `MkWrapped :: Int -> WrappedInt`

### Actual behaviour
The tooltip shows the type `MkWrapped :: Int %1 -> WrappedInt`, which includes linear arrows.

**Comment by michaelpj** [comment]
cc @July541, who looked a pretty-printing of stuff recently?

**Comment by July541** [comment]
A quick peek: we rely on `printOutputable` to print type info, I tried `-fprint-explicit-runtime-reps` suggested in the [guide](-fprint-explicit-runtime-reps) but not work. I'll dive into it later.

https://github.com/haskell/haskell-language-server/blob/b1bf5499155e259341e4868863b0fd743b6bd65f/ghcide/src/Development/IDE/Spans/AtPoint.hs#L249-L252

**Comment by July541** [comment]
The test is inconsistent with the actual behavior, odd.
Test:
<img width="1064" alt="image" src="https://user-images.githubusercontent.com/16057632/168439495-8ce89161-6caf-4514-bac6-f9e2b39f7b99.png">

Actual:
<img width="804" alt="image" src="https://user-images.githubusercontent.com/16057632/168439505-811bf640-1a88-4894-b622-eed387d817f4.png">

**Comment by July541** [comment]
It's from https://github.com/haskell/haskell-language-server/blob/b1bf5499155e259341e4868863b0fd743b6bd65f/ghcide/src/Development/IDE/Spans/AtPoint.hs#L232-L238

`maybeKind` if we have `newtype WrappedInt = MkWrapped Int` and hover on `MkWrapped`. And it is something about `(AConLike (RealDataCon dataCon))`
https://github.com/haskell/haskell-language-server/blob/b1bf5499155e259341e4868863b0fd743b6bd65f/ghcide/src/Development/IDE/Spans/Common.hs#L40-L49

We will get linear arrow If we print `RealDataCon` directly, and it can be disabled by [`dataConNonlinearType`](https://hackage.haskell.org/package/ghc-9.2.1/docs/GHC-Core-DataCon.html#v:dataConNonlinearType).

I Don't know how to control the printing in `showSDoc`, A potential solution is reading the dynflag about lineartype to control the behavior by `dataConNonlinearType`, not sure there exists a better way.

**Comment by michaelpj** [comment]
Is this still a problem?

**Comment by jhrcek** [comment]
Yes, I’m pretty sure I saw it today with his built from master, using ghc
9.4.8

**Comment by emlautarom1** [comment]
I can reproduce the issue on GHC 9.4.8 + HLS 2.7.0.0
